packages/flutter/lib/src/material/button.dart

```diff
@@ -255,12 +259,14 @@ class _MaterialButtonState extends State<MaterialButton> {
     final ButtonTheme buttonTheme = ButtonTheme.of(context);
     final double height = config.height ?? buttonTheme.height;
     final int elevation = (_highlight ? config.highlightElevation : config.elevation) ?? 0;
+    final bool hasColorOrElevation = (config.color != null || elevation > 0);
     Widget contents = new IconTheme.merge(
       context: context,
       data: new IconThemeData(
         color: textColor
       ),
       child: new InkWell(
+        borderRadius: hasColorOrElevation ? BorderRadius.zero : kMaterialButtonDefaultBorderRadius,
```

Comment:
No. The existing behaviour was that the `Material` wrapper will only be added to the tree if there is a color or elevation and that will clip the `InkWell` for us so we don't double clip.